specs/governance/liveness-module.md

```diff
@@ -0,0 +1,216 @@
+
+### Multisig Safety Failure
+
+A multisig is considered to be in a safety failure state if the number of active malicious owners is equal or greater
+than `quorum`.
+
+### Fallback Owner
+
+The owner that is appointed to take control of a multisig in case of a liveness or safety failure.
+
+### Liveness Challenge
+
+A challenge to a multisig to prove that it is not in a liveness failure state.
+
+## Assumptions
+
+### aLM-001: The Fallback Owner is Honest
+
+The fallback owner is chosen by the multisig which can verify that it is honest.
+
+#### Severity: Medium to High
+
+If this assumption is false and not known to be false, the LivenessModule is not able to recover the multisig from a
+liveness failure state until a new fallback owner is chosen.
+
+If this assumption is false but not known to be false, and a challenge is successful, the multisig would enter into a
+safety failure state.
+
+### aLM-002: The Fallback Owner is Active
+
+The fallback owner is assumed to be active.
+
+#### Severity: High
+
+If this assumption is false, the LivenessModule is not operational and if the multisig would fall into a liveness
+failure state it would not be able to recover.
+
+## Invariants
+
+### iLM-001: No Concurrent Challenges
+
+For an enabled `safe`, there can't be more than one concurrent challenge.
+
+#### Severity: Medium
+
+If this invariant is broken, an attacker could spam the multisig with challenges, damaging its operational performance.
+
+### iLM-002: Honest Users Can Recover From Temporary Key Control Over a Quorum of Keys
+
+If an attacker has full, joint, or temporary key control over less than a quorum of keys, honest users should always
+be able to recover the account by transferring ownership to the fallback owner
+
+#### Severity: High
+
+If this invariant is broken, an attacker with temporary key control over less than a quorum of keys could force the
+multisig into a liveness failure state.
+
+### iLM-003: A Quorum Of Honest Users Retains Ownership
+
+While a quorum of honest users exist, they should remain in control of the account.
+
+#### Severity: Medium
+
+If this invariant is broken, there would be an operational and possibly reputational impact while the ownership of the
+account is restablished to the account owners.
+
+## Function Specification
+
+### `enableModule`
+
+Enables the module by the multisig to be challenged and sets the `liveness_challenge_period` and `fallback_owner`.
+
+- MUST set the caller as a `safe`.
+- MUST allow an arbitrary number of `safe` contracts to use the module.
+- MUST take as parameters `liveness_challenge_period` and `fallback_owner` and store them as related to the `safe`.
+
+### `disableModule`
+
+Disables the module by an enabled `safe`.
+
+- MUST only be executable an enabled `safe`.
+- MUST erase the existing `liveness_challenge_period` and `fallback_owner` data related to the calling `safe`.
```

Comment:
is `challenge_start_time` used to determine whether a challenge is active? what happens to it here?